centos_ci.sh

```diff
@@ -22,79 +22,155 @@ set -e
+  GLIDE_TAG=v0.12.3
+  GLIDE_LATEST_RELEASE_URL="https://github.com/Masterminds/glide/releases/download/${GLIDE_TAG}/glide-${GLIDE_TAG}-${GLIDE_OS_ARCH}.tar.gz"
+  curl -LO ${GLIDE_LATEST_RELEASE_URL}
+  mkdir /tmp/glide
+  tar --directory=/tmp/glide -xvf glide-${GLIDE_TAG}-${GLIDE_OS_ARCH}.tar.gz
+  export PATH=$PATH:/tmp/glide/${GLIDE_OS_ARCH}
+}
+
+function prepare() {
+  install_and_setup_golang;
+  setup_repo;
+  setup_glide;
+
+  echo "CICO: Preparation complete"
+}
+
+function run_tests() {
+  cd $GOPATH/src/github.com/minishift/minishift
+  make clean test cross fmtcheck prerelease
+  # Run integration test with 'kvm' driver
+  MINISHIFT_VM_DRIVER=kvm make integration
+
+  echo "CICO: Tests ran successfully"
+}
+
+function artifacts_upload_on_pr_and_master_trigger() {
+  set +x
+  # For PR build, GIT_BRANCH is set to branch name other than origin/master
+  if [[ "$GIT_BRANCH" = "origin/master" ]]; then
+    # http://stackoverflow.com/a/22908437/1120530; Using --relative as --rsync-path not working
+    mkdir -p minishift/master/$BUILD_NUMBER/
+    cp -r out/* minishift/master/$BUILD_NUMBER/
+    RSYNC_PASSWORD=$1 rsync -a --relative minishift/master/$BUILD_NUMBER/ minishift@artifacts.ci.centos.org::minishift/
+    echo "Find Artifacts here http://artifacts.ci.centos.org/minishift/minishift/master/$BUILD_NUMBER ."
+  else
+    # http://stackoverflow.com/a/22908437/1120530; Using --relative as --rsync-path not working
+    mkdir -p minishift/pr/$ghprbPullId/
+    cp -r out/* minishift/pr/$ghprbPullId/
+    RSYNC_PASSWORD=$1 rsync -a --relative minishift/pr/$ghprbPullId/ minishift@artifacts.ci.centos.org::minishift/
+    echo "Find Artifacts here http://artifacts.ci.centos.org/minishift/minishift/pr/$ghprbPullId ."
+  fi
+}
+
+function docs_artifacts_upload() {
+  set +x
+
+  version=$(cat docs/build/variables.adoc | cut -d' ' -f2)
+  mkdir -p minishift/docs/$(version)
+  cp docs/build/minishift-adocs.tar minishift/docs/$(version)/
+  ln -s $(pwd)/minishift/docs/$(version) minishift/docs/latest-alpha # TODO: make it to latest when everything works
   # http://stackoverflow.com/a/22908437/1120530; Using --relative as --rsync-path not working
-  mkdir -p minishift/master/$BUILD_NUMBER/
-  cp -r out/* minishift/master/$BUILD_NUMBER/
-  RSYNC_PASSWORD=$PASS rsync -a --relative minishift/master/$BUILD_NUMBER/ minishift@artifacts.ci.centos.org::minishift/
-  echo "Find Artifacts here http://artifacts.ci.centos.org/minishift/minishift/master/$BUILD_NUMBER ."
+  RSYNC_PASSWORD=$1 rsync -a --relative minishift/docs minishift@artifacts.ci.centos.org::minishift/
+  echo "Find docs artifacts here http://artifacts.ci.centos.org/minishift/minishift/docs/$(version)."
+}
+
+# Execution begins here
+load_jenkins_vars;
+
+# Intentionally using wrong condition to test docs artifacts uploads
+if [[ "$JOB_NAME" != "minishift-docs" ]]; then
+  prepare_docs_user;
+  # Run everything beyond this point as 'docs' user
+  runuser -l docs -c install_core_deps;
```

Comment:
What's the state on this one? I was thinking about this a bit more, if possible I'd like to run all builds as a new/created user. This way we can extend the master build as well. On a push to master we build the docs tarball as well and then use it in a checkout of docs.openshift.org to build the full docs site which we then can stage on the archive server.